pytests/func/test_django.py

```diff
@@ -0,0 +1,190 @@
+
+    # connect to web server
+    with get_web_content(link, {}) as web_result:
+        thread_stopped = debug_session.wait_for_next(Event('stopped', ANY.dict_with({'reason': 'breakpoint'})))
+        assert thread_stopped.body['threadId'] is not None
+
+        tid = thread_stopped.body['threadId']
+
+        resp_stacktrace = debug_session.send_request('stackTrace', arguments={
+            'threadId': tid,
+        }).wait_for_response()
+        assert resp_stacktrace.body['totalFrames'] > 1
+        frames = resp_stacktrace.body['stackFrames']
+        assert frames[0] == {
+            'id': 1,
+            'name': bp_name,
+            'source': {
+                'sourceReference': ANY,
+                'path': bp_file,
+            },
+            'line': bp_line,
+            'column': 1,
+        }
+
+        fid = frames[0]['id']
+        resp_scopes = debug_session.send_request('scopes', arguments={
+            'frameId': fid
+        }).wait_for_response()
+        scopes = resp_scopes.body['scopes']
+        assert len(scopes) > 0
+
+        resp_variables = debug_session.send_request('variables', arguments={
+            'variablesReference': scopes[0]['variablesReference']
+        }).wait_for_response()
+        variables = list(v for v in resp_variables.body['variables'] if v['name'] == 'content')
+        assert variables == [{
+                'name': 'content',
+                'type': 'str',
+                'value': repr(bp_var_content),
+                'presentationHint': {'attributes': ['rawString']},
+                'evaluateName': 'content'
+            }]
+
+        debug_session.send_request('continue').wait_for_response()
+        debug_session.wait_for_next(Event('continued'))
+
+    assert web_result['content'].find(bp_var_content) != -1
+
+    # shutdown to web server
+    link += 'exit' if link.endswith('/') else '/exit'
+    with get_web_content(link):
+        pass
+
+def test_django_breakpoint_no_multiproc(debug_session):
+    _django_breakpoint_no_multiproc(debug_session, DJANGO1_MANAGE, 40, 'home')
+
+def test_django_template_breakpoint_no_multiproc(debug_session):
+    _django_breakpoint_no_multiproc(debug_session, DJANGO1_TEMPLATE, 8, 'Django Template')
+
+
+def _django_exception_no_multiproc(debug_session, ex_type, ex_line):
+    _django_no_multiproc_common(debug_session)
+    debug_session.prepare_to_run(filename=DJANGO1_MANAGE)
+
+    debug_session.send_request('setExceptionBreakpoints', arguments={
+        'filters': ['raised', 'uncaught'],
+    }).wait_for_response()
+
+    debug_session.start_debugging()
+
+    base_link = _wait_for_django_link(debug_session)
+    assert base_link is not None
+
+    link = base_link + ex_type if base_link.endswith('/') else ('/' + ex_type)
+    with get_web_content(link, {}):
+        thread_stopped = debug_session.wait_for_next(Event('stopped', ANY.dict_with({'reason': 'exception'})))
+        assert thread_stopped == Event('stopped', ANY.dict_with({
```

Comment:
Does it need `ANY` here - i.e. is there's anything else in the dict?

Reply:
yes, there is text, description, they can have value depending on the situation.

Reply:
But it does explicitly check for text and description.

Reply:
sorry that would be `preserveFocusHint`, this can be True or False, also `allThreadsStopped` can be true or false. After taking fabio's change we will have a consistent value for these.

Reply:
Ah, I see.

Even after the change, we probably don't want this to check everything - after all, it isn't really meant to be a test of exceptions, and we'll have a separate test for those that does actually check every entry in the dict. For this test, we really only care that an exception happened with a particular message, and that's exactly what the body checks right now, so it's good!